xclim/indices/generic.py

```diff
@@ -290,3 +295,303 @@ def daily_downsampler(da: xr.DataArray, freq: str = "YS") -> xr.DataArray:
 
     # return groupby according to tags
     return buffer.groupby("tags")
+
+
+# CF-INDEX-META Indices
+
+
+def count_level_crossings(
+    low_data: xr.DataArray, high_data: xr.DataArray, threshold: str, freq: str
+):
+    """This index function takes two inputs, low_data and high_data, together with one parameter, the threshold.
+
+    It calculates the number of times low_data is below threshold while high_data is above threshold.
+    First, the threshold is transformed to the same standard_name and units as the input data,
+    then the thresholding is performed, and finally, the number of occurrences is counted.
+
+    Parameters
+    ----------
+    low_data: xr.DataArray
+    high_data: xr.DataArray
+    threshold: str
+      Quantity.
+    freq: str
+      Resampling frequency.
+    """
+    # Convert units to low_data
+    high_data = convert_units_to(high_data, low_data)
+    threshold = convert_units_to(threshold, low_data)
+
+    lower = compare(low_data, "<", threshold)
+    higher = compare(high_data, ">", threshold)
+
+    out = (lower & higher).resample(time=freq).sum()
+    out.attrs["units"] = ""
+    return out
+
+
+def count_occurrences(data: xr.DataArray, threshold: str, condition: str, freq: str):
+    """Calculates the number of times some condition is met.
+
+    First, the threshold is transformed to the same standard_name and units as the input data.
+    Then the thresholding is performed as condition(data, threshold),
+    i.e. if condition is <, data < threshold.
+    Finally, count the number of occurrences when condition is met.
+
+    Parameters
+    ----------
+    data : xr.DataArray
+    threshold : str
+      Quantity.
+    condition : {">", "<", ">=", "<=", "==", "!="}
+      Operator.
+    freq: str
+      Resampling frequency.
+    """
+    threshold = convert_units_to(threshold, data)
+
+    cond = compare(data, condition, threshold)
+
+    out = cond.resample(time=freq).sum()
+    out.attrs["units"] = ""
+    return out
+
+
+def diurnal_temperature_range(
+    low_data: xr.DataArray, high_data: xr.DataArray, freq: str
+):
+    """This index function takes two inputs, low_data and high_data, i.e. daily minimum and maximum temperature.
+
+    From this it calculated the average diurnal temperature range.
+
```

Comment:
Use NumPy doc standard for docstring. First line should say what the function does.